src/fetch-mocker.js

```diff
@@ -472,6 +538,118 @@ export class FetchMocker {
 		return preflightData;
 	}
 
+	/**
+	 * Processes a redirect response according to the fetch spec
+	 * @param {Response} response The response to check for a redirect
+	 * @param {Request} request The original request
+	 * @param {URL[]} urlList The list of URLs already visited in this redirect chain
+	 * @param {any} requestBody The body of the original request
+	 * @returns {Promise<Response>} The final response after any redirects
+	 */
+	async #processRedirect(response, request, urlList = [], requestBody = null) {
+		// Add current URL to list
+		urlList.push(new URL(request.url));
+		
+		const isRedirect = isRedirectStatus(response.status);
+
+		// Process response based on redirect status and mode
+		if (!isRedirect) {
+			// Not a redirect - set redirected flag if we've had previous redirects
+			if (urlList.length > 1) {
+				Object.defineProperty(response, "redirected", { value: true });
+			}
+			return response;
+		}
+		
+		// Handle based on redirect mode
+		switch (request.redirect) {
+			case "manual":
+				// Return an opaque redirect response
+				return createOpaqueRedirectResponse(this.#Response, request.url);
+				
+			case "error":
+				// Just throw an error
+				throw new TypeError(`Redirect at ${request.url} was blocked due to redirect mode being 'error'`);
+				
+			case "follow":
+			default:
+				// Continue with redirect handling
+				break;
+		}
+		
+		// Get and validate the redirect location
+		const location = response.headers.get("Location");
+		if (!location) {
+			throw new TypeError(`Redirect at ${request.url} has no Location header`);
+		}
+
+		// Construct the new URL
+		let redirectUrl;
+		try {
+			redirectUrl = new URL(location, request.url);
+		} catch {
+			throw new TypeError(`Invalid redirect URL: ${location}`);
+		}
+
+		// Check for redirect loops
+		if (urlList.some(url => url.href === redirectUrl.href)) {
+			throw new TypeError(`Redirect loop detected for ${redirectUrl.href}`);
+		}
+
+		// Check redirect limit
+		if (urlList.length >= 20) {
+			throw new TypeError("Too many redirects (maximum is 20)");
+		}
+
+		// Create a new request for the redirect
```

Comment:
Somewhere in here, we need to strip these headers:
https://fetch.spec.whatwg.org/#request-body-header-name